app/components/admin/custom_fields/custom_field_projects/row_component.rb

```diff
@@ -37,8 +37,31 @@ def wrapper_uniq_by
         end
 
         def more_menu_items
-          []
+          @more_menu_items ||= [more_menu_detach_project].compact
         end
+
+        private
+
+        def more_menu_detach_project
+          if User.current.admin && project.active?
```

Comment:
You could also query for `User.current.allowed_to?(:select_custom_fields, project)`. This is more specific than the admin check (only admins will visit that page anyway), but allows to later on open up the page to non admins more easily. At the same time, it will check for the project being active as no permissions are granted on archived projects.

Reply:
Good call! 👍🏾